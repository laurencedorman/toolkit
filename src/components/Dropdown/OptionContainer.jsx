/**
 * Copyright (c) Colibri SAS - ManoMano
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import ToggleDown from '../ToggleDown';
import styles from './Dropdown.module.scss';

/**
 * @visibleName Dropdown
 */
const OptionContainer = ({
  on,
  toggle,
  size,
  options,
  right,
  className,
  onClick,
  active,
  sideLeft,
  sideRight,
}) => {
  const container = cn(styles.container, {
    [styles.show]: on,
    [styles.right]: right,
    [styles.sideLeft]: sideLeft,
    [styles.sideRight]: sideRight,
  });

  const itemOption = item =>
    cn(
      'li',
      className,
      { [styles.disabled]: item.disabled },
      { [styles.active]: (item.value || item.id) === active }
    );

  const list = cn('ul', styles.list);

  const space = 8;

  const setPosition = {
    top: size && !sideLeft && !sideRight && `${size.height + space}px`,
    right: size && sideLeft && !sideRight && `${size.width + space}px`,
    left: size && sideRight && `${size.width + space}px`,
  };

  return (
    <>
      {on && <div className={styles.closeTarget} onClick={toggle} />}
      <ToggleDown on={on} className={container} style={setPosition}>
        <ul className={list}>
          {options.map((item, index) => {
            const key = item.id ? item.id.toString() : index;
            return (
              <li
                key={key}
                className={itemOption(item)}
                data-id={item.id}
                data-value={item.value}
                disabled={item.disabled}
                onClick={!item.disabled ? onClick : null}
                onKeyPress={null}
              >
                <span
                  onClick={!item.disabled ? toggle : null}
                  onKeyPress={null}
                  key={key}
                >
                  {item.value}
                </span>
              </li>
            );
          })}
        </ul>
      </ToggleDown>
    </>
  );
};

OptionContainer.propTypes = {
  on: PropTypes.bool.isRequired,
  options: PropTypes.arrayOf(PropTypes.shape()).isRequired,
  toggle: PropTypes.func,
  right: PropTypes.bool,
  onClick: PropTypes.func,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  active: PropTypes.string,
  sideLeft: PropTypes.bool,
  sideRight: PropTypes.bool,
};

OptionContainer.defaultProps = {
  toggle: null,
  onClick: null,
  right: false,
  className: '',
  disabled: false,
  active: '',
  sideLeft: false,
  sideRight: false,
};

export default OptionContainer;
