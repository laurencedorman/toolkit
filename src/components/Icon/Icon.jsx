// @flow
import React from 'react';
import cn from 'classnames';

import iconAssets from '../../assets/setIcons';
import styles from './Icon.module.scss';

/**
 * @visibleName Icon
 */
const IconComponent = ({
  onClick, className, name, size, fill, dataQa,
}: propTypes) => {
  const classNames = cn(
    styles.container,
    className,
  );

  return (
    <div
      className={classNames}
      onClick={onClick}
      onKeyPress={onClick}
      tabIndex="-1"
      role="button"
      data-qa={dataQa}
    >
      {Object.prototype.hasOwnProperty.call(iconAssets, name)
      && (
        <svg
          className={styles.svg}
          width={`${String(size)}px`}
          height={`${String(size)}px`}
          viewBox={iconAssets[name].viewBox}
          aria-describedby={name}
          fill={fill}
          xmlns="http://www.w3.org/2000/svg"
          role="img"
        >
          <path
            d={iconAssets[name].d}
          />
        </svg>
      )}
    </div>
  );
};

type propTypes = {
  onClick?: () => void,
  className?: string,
  name: string,
  size?: number,
  fill?: string | () => void,
  dataQa?: string,
};

IconComponent.defaultProps = {
  onClick: null,
  className: '',
  size: 32,
  fill: '#0c193a',
  dataQa: '',
};

export default IconComponent;
