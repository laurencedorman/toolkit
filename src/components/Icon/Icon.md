Icon:

- Import component icon with props <b>component</b>.
- You need to import Icon component and IconName.
- Icon was in SVG.
- Open a Merge Request if you need new icon.

  <br />

Import example:

```jsx static
import { Icon, IconAccount } from 'ToolkitPath';

<Icon component={IconAccount} />;
```

```js
import IconAbTest from './Icons/IconAbTest';
import IconAccount from './Icons/IconAccount';
import IconAlertCircle from './Icons/IconAlertCircle';
import IconAnalyse from './Icons/IconAnalyse';
import IconArrowFullDown from './Icons/IconArrowFullDown';
import IconArrowFullLeft from './Icons/IconArrowFullLeft';
import IconArrowFullRight from './Icons/IconArrowFullRight';
import IconArrowFullUp from './Icons/IconArrowFullUp';
import IconAward from './Icons/IconAward';
import IconBankTransfert from './Icons/IconBankTransfert';
import IconBarredEye from './Icons/IconBarredEye';
import IconBasket from './Icons/IconBasket';
import IconBathTub from './Icons/IconBathTub';
import IconBigDrill from './Icons/IconBigDrill';
import IconBox from './Icons/IconBox';
import IconBoxes from './Icons/IconBoxes';
import IconBrush from './Icons/IconBrush';
import IconBurgerMenu from './Icons/IconBurgerMenu';
import IconCalendar from './Icons/IconCalendar';
import IconCamera from './Icons/IconCamera';
import IconCheck from './Icons/IconCheck';
import IconChevronDown from './Icons/IconChevronDown';
import IconChevronLeft from './Icons/IconChevronLeft';
import IconChevronRight from './Icons/IconChevronRight';
import IconChevronUp from './Icons/IconChevronUp';
import IconCircleMinus from './Icons/IconCircleMinus';
import IconCirclePlus from './Icons/IconCirclePlus';
import IconClock from './Icons/IconClock';
import IconClose from './Icons/IconClose';
import IconCloseCircle from './Icons/IconCloseCircle';
import IconContent from './Icons/IconContent';
import IconCreditCard from './Icons/IconCreditCard';
import IconDeliveryForm from './Icons/IconDeliveryForm';
import IconDocument from './Icons/IconDocument';
import IconDotMenu from './Icons/IconDotMenu';
import IconDownload from './Icons/IconDownload';
import IconDragNdrop from './Icons/IconDragNdrop';
import IconDrill from './Icons/IconDrill';
import IconEdit from './Icons/IconEdit';
import IconExtra from './Icons/IconExtra';
import IconEye from './Icons/IconEye';
import IconFacebook from './Icons/IconFacebook';
import IconFaucetWater from './Icons/IconFaucetWater';
import IconFeatureFlag from './Icons/IconFeatureFlag';
import IconFilter from './Icons/IconFilter';
import IconFlash from './Icons/IconFlash';
import IconFreeDelivery from './Icons/IconFreeDelivery';
import IconHeart from './Icons/IconHeart';
import IconHeartFull from './Icons/IconHeartFull';
import IconHelpCircle from './Icons/IconHelpCircle';
import IconHistoricalSearch from './Icons/IconHistoricalSearch';
import IconHood from './Icons/IconHood';
import IconInformationCircle from './Icons/IconInformationCircle';
import IconInstagram from './Icons/IconInstagram';
import IconLabel from './Icons/IconLabel';
import IconLamp from './Icons/IconLamp';
import IconLaunch from './Icons/IconLaunch';
import IconLink from './Icons/IconLink';
import IconLock from './Icons/IconLock';
import IconLogOut from './Icons/IconLogOut';
import IconMail from './Icons/IconMail';
import IconMapPin from './Icons/IconMapPin';
import IconMarketing from './Icons/IconMarketing';
import IconMessage from './Icons/IconMessage';
import IconMessageCircle from './Icons/IconMessageCircle';
import IconPhone from './Icons/IconPhone';
import IconPiggyBank from './Icons/IconPiggyBank';
import IconPinterest from './Icons/IconPinterest';
import IconPlay from './Icons/IconPlay';
import IconPrinter from './Icons/IconPrinter';
import IconProduct from './Icons/IconProduct';
import IconPromo from './Icons/IconPromo';
import IconScrew from './Icons/IconScrew';
import IconSearch from './Icons/IconSearch';
import IconStarEmpty from './Icons/IconStarEmpty';
import IconStarFull from './Icons/IconStarFull';
import IconStop from './Icons/IconStop';
import IconTagPromo from './Icons/IconTagPromo';
import IconThumbDown from './Icons/IconThumbDown';
import IconTrash from './Icons/IconTrash';
import IconTrending from './Icons/IconTrending';
import IconTrucks from './Icons/IconTrucks';
import IconTwitter from './Icons/IconTwitter';
import IconVote from './Icons/IconVote';
import IconWall from './Icons/IconWall';
import IconWateringCan from './Icons/IconWateringCan';
import IconWheel from './Icons/IconWheel';
import IconYoutube from './Icons/IconYoutube';

import IconBelgium from './Icons/IconBelgium';
import IconFrance from './Icons/IconFrance';
import IconGermany from './Icons/IconGermany';
import IconItaly from './Icons/IconItaly';
import IconNetherlands from './Icons/IconNetherlands';
import IconSpain from './Icons/IconSpain';
import IconSwitzerland from './Icons/IconSwitzerland';
import IconUnitedKingdom from './Icons/IconUnitedKingdom';

const icons = [
  { icon: IconAbTest, name: 'IconAbTest' },
  { icon: IconAccount, name: 'IconAccount' },
  { icon: IconAlertCircle, name: 'IconAlertCircle' },
  { icon: IconAnalyse, name: 'IconAnalyse' },
  { icon: IconArrowFullDown, name: 'IconArrowFullDown' },
  { icon: IconArrowFullLeft, name: 'IconArrowFullLeft' },
  { icon: IconArrowFullRight, name: 'IconArrowFullRight' },
  { icon: IconArrowFullUp, name: 'IconArrowFullUp' },
  { icon: IconAward, name: 'IconAward' },
  { icon: IconBankTransfert, name: 'IconBankTransfert' },
  { icon: IconBarredEye, name: 'IconBarredEye' },
  { icon: IconBasket, name: 'IconBasket' },
  { icon: IconBathTub, name: 'IconBathTub' },
  { icon: IconBigDrill, name: 'IconBigDrill' },
  { icon: IconBox, name: 'IconBox' },
  { icon: IconBoxes, name: 'IconBoxes' },
  { icon: IconBrush, name: 'IconBrush' },
  { icon: IconBurgerMenu, name: 'IconBurgerMenu' },
  { icon: IconCalendar, name: 'IconCalendar' },
  { icon: IconCamera, name: 'IconCamera' },
  { icon: IconCheck, name: 'IconCheck' },
  { icon: IconChevronDown, name: 'IconChevronDown' },
  { icon: IconChevronLeft, name: 'IconChevronLeft' },
  { icon: IconChevronRight, name: 'IconChevronRight' },
  { icon: IconChevronUp, name: 'IconChevronUp' },
  { icon: IconCircleMinus, name: 'IconCircleMinus' },
  { icon: IconCirclePlus, name: 'IconCirclePlus' },
  { icon: IconClock, name: 'IconClock' },
  { icon: IconClose, name: 'IconClose' },
  { icon: IconCloseCircle, name: 'IconCloseCircle' },
  { icon: IconContent, name: 'IconContent' },
  { icon: IconCreditCard, name: 'IconCreditCard' },
  { icon: IconDeliveryForm, name: 'IconDeliveryForm' },
  { icon: IconDocument, name: 'IconDocument' },
  { icon: IconDotMenu, name: 'IconDotMenu' },
  { icon: IconDownload, name: 'IconDownload' },
  { icon: IconDragNdrop, name: 'IconDragNdrop' },
  { icon: IconDrill, name: 'IconDrill' },
  { icon: IconEdit, name: 'IconEdit' },
  { icon: IconExtra, name: 'IconExtra' },
  { icon: IconEye, name: 'IconEye' },
  { icon: IconFacebook, name: 'IconFacebook' },
  { icon: IconFaucetWater, name: 'IconFaucetWater' },
  { icon: IconFeatureFlag, name: 'IconFeatureFlag' },
  { icon: IconFilter, name: 'IconFilter' },
  { icon: IconFlash, name: 'IconFlash' },
  { icon: IconFreeDelivery, name: 'IconFreeDelivery' },
  { icon: IconHeart, name: 'IconHeart' },
  { icon: IconHeartFull, name: 'IconHeartFull' },
  { icon: IconHelpCircle, name: 'IconHelpCircle' },
  { icon: IconHistoricalSearch, name: 'IconHistoricalSearch' },
  { icon: IconHood, name: 'IconHood' },
  { icon: IconInformationCircle, name: 'IconInformationCircle' },
  { icon: IconInstagram, name: 'IconInstagram' },
  { icon: IconLabel, name: 'IconLabel' },
  { icon: IconLamp, name: 'IconLamp' },
  { icon: IconLaunch, name: 'IconLaunch' },
  { icon: IconLink, name: 'IconLink' },
  { icon: IconLock, name: 'IconLock' },
  { icon: IconLogOut, name: 'IconLogOut' },
  { icon: IconMail, name: 'IconMail' },
  { icon: IconMapPin, name: 'IconMapPin' },
  { icon: IconMarketing, name: 'IconMarketing' },
  { icon: IconMessage, name: 'IconMessage' },
  { icon: IconMessageCircle, name: 'IconMessageCircle' },
  { icon: IconPhone, name: 'IconPhone' },
  { icon: IconPiggyBank, name: 'IconPiggyBank' },
  { icon: IconPinterest, name: 'IconPinterest' },
  { icon: IconPlay, name: 'IconPlay' },
  { icon: IconPrinter, name: 'IconPrinter' },
  { icon: IconProduct, name: 'IconProduct' },
  { icon: IconPromo, name: 'IconPromo' },
  { icon: IconScrew, name: 'IconScrew' },
  { icon: IconSearch, name: 'IconSearch' },
  { icon: IconStarEmpty, name: 'IconStarEmpty' },
  { icon: IconStarFull, name: 'IconStarFull' },
  { icon: IconStop, name: 'IconStop' },
  { icon: IconTagPromo, name: 'IconTagPromo' },
  { icon: IconThumbDown, name: 'IconThumbDown' },
  { icon: IconTrash, name: 'IconTrash' },
  { icon: IconTrending, name: 'IconTrending' },
  { icon: IconTrucks, name: 'IconTrucks' },
  { icon: IconTwitter, name: 'IconTwitter' },
  { icon: IconVote, name: 'IconVote' },
  { icon: IconWall, name: 'IconWall' },
  { icon: IconWateringCan, name: 'IconWateringCan' },
  { icon: IconWheel, name: 'IconWheel' },
  { icon: IconYoutube, name: 'IconYoutube' },
  { icon: IconBelgium, name: 'IconBelgium' },
  { icon: IconFrance, name: 'IconFrance' },
  { icon: IconGermany, name: 'IconGermany' },
  { icon: IconItaly, name: 'IconItaly' },
  { icon: IconNetherlands, name: 'IconNetherlands' },
  { icon: IconSpain, name: 'IconSpain' },
  { icon: IconSwitzerland, name: 'IconSwitzerland' },
  { icon: IconUnitedKingdom, name: 'IconUnitedKingdom' },
];

const style = {
  display: 'flex',
  flexFlow: 'column wrap',
  justifyContent: 'center',
  alignItems: 'center',
  width: 140,
  height: 80,
  padding: '.5rem',
  margin: '0 .5rem .5rem 0',
  background: '#e8e8e8',
  border: '1px solid #d4d4d4',
  borderRadius: 3,
};

const span = {
  marginTop: 5,
  fontSize: 12,
  textAlign: 'center',
};

<div
  style={{
    display: 'flex',
    flexFlow: 'row wrap',
    justifyContent: 'flex-start',
  }}
>
  {icons.map(({ icon, name }) => (
    <div key={name} style={style}>
      <Icon component={icon} />
      <span style={span}>{name}</span>
    </div>
  ))}
</div>;
```
