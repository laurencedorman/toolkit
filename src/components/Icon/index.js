export { default } from './Icon';
export { default as Svg } from './Svg';
export { default as Img } from './Img';
