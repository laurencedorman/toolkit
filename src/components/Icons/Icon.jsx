// @flow
import React from 'react';
import styles from './Icon.module.scss';
import icons from '../../assets/setIcons';

const Icon = ({ name, size, fill }: propTypes) => (
  <svg
    className={styles.svg}
    width={`${size}px`}
    height={`${size}px`}
    viewBox={icons[name].viewBox}
    aria-labelledby={icons[name]}
    xmlns="http://www.w3.org/2000/svg"
    role="img">
    <path
      d={icons[name].d}
      fill={fill} />
  </svg>
);

type propTypes = {
  name: string,
  size?: number,
  fill?: string,
};

Icon.defaultProps = {
  size: 32,
  fill: '#0c193a',
};

export default Icon;
