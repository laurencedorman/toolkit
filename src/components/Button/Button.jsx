// @flow
import React from 'react';
import Icon from '../Icon';
import cn from 'classnames';
import styles from './Button.module.scss';

/**
 * @visibleName Button
 */
const Button = ({
  children, size, theme, disabled, type, icon, reverse, onClick, dataQa }: propTypes) => {
  const className = cn(
    styles.manoolkitButton,
    styles[size],
    styles[theme],
    {
      [styles.icon]: icon,
      [styles.reverse]: reverse,
      [styles.onlyIcon]: !children,
    }
  );

  return(
    <button
      disabled={disabled}
      type={type}
      onClick={onClick}
      className={className}
      data-qa={dataQa}
    >
      {icon &&
      <Icon name={icon} size="16" />}
      {children && children}
    </button>
  );
};

type propTypes = {
  children?: string | Node,
  theme?: 'default' | 'primary' | 'secondary',
  size?: 'big' | 'default' | 'small' | 'xsmall',
  type?: 'submit' | 'reset' | 'button' | 'menu',
  icon?: string,
  reverse?: boolean,
  disabled?: boolean,
  onClick?: () => void,
  dataQa?: string,
};

Button.defaultProps = {
  children: null,
  theme: 'default',
  size: 'default',
  type: 'button',
  icon: '',
  reverse: false,
  disabled: false,
  onClick: null,
  dataQa: '',
};

export default Button;
