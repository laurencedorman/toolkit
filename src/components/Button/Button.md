Default Button:

```js
import Button from '../Button';
import Layout from '../Layout';

<Layout styleGuidist direction="row">
  <Button>default Button</Button>
  <Button theme="primary">primary Button</Button>
  <Button theme="secondary">secondary Button</Button>
  <Button theme="ghost">ghost Button</Button>
  <Button theme="reset">reset style Button</Button>
  <Button theme="link">Link Button</Button>
  <Button disabled>disabled Button</Button>
</Layout>;
```

Button Size:

```js
import Button from '../Button';
import Layout from '../Layout';

<Layout styleGuidist direction="row">
  <Button size="big">big button</Button>
  <Button>default button</Button>
  <Button size="small">small button</Button>
  <Button size="xsmall">xsmall button</Button>
</Layout>;
```

Icon Button:

```js
import Button from '../Button';
import IconEdit from '../Icon/Icons/IconEdit';
import Layout from '../Layout';

<Layout styleGuidist direction="row">
  <Button size="big" iconComponent={IconEdit}>
    Button
  </Button>
  <Button iconComponent={IconEdit}>Button</Button>
  <Button size="small" iconComponent={IconEdit}>
    Button
  </Button>
  <Button size="xsmall" iconComponent={IconEdit}>
    Button
  </Button>
</Layout>;
```

Icon Reverse Button:

```js
import Button from '../Button';
import IconEdit from '../Icon/Icons/IconEdit';
import Layout from '../Layout';

<Layout styleGuidist direction="row">
  <Button theme="primary" size="big" iconComponent={IconEdit} reverse>
    Button
  </Button>
  <Button theme="primary" iconComponent={IconEdit} reverse>
    Button
  </Button>
  <Button theme="primary" size="small" iconComponent={IconEdit} reverse>
    Button
  </Button>
  <Button theme="primary" size="xsmall" iconComponent={IconEdit} reverse>
    Button
  </Button>
</Layout>;
```

Icon only Button:

```js
import Button from '../Button';
import IconEdit from '../Icon/Icons/IconEdit';
import Layout from '../Layout';

<Layout styleGuidist direction="row">
  <Button size="big" iconComponent={IconEdit} />
  <Button iconComponent={IconEdit} />
  <Button iconComponent={IconEdit} theme="reset" fill="#424242" />
  <Button size="small" iconComponent={IconEdit} />
  <Button size="xsmall" iconComponent={IconEdit} />
</Layout>;
```
