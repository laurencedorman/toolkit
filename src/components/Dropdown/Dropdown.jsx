/**
 * Copyright (c) Colibri SAS - ManoMano
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

import Button from '../Button';
import Icon from '../Icon';
import ChevronRight from '../Icon/Icons/IconChevronRight';
import { useMeasure } from '../GetMeasure';
import OptionContainer from './OptionContainer';

import colors from '../../styles/colors';
import styles from './Dropdown.module.scss';

const Dropdown = ({
  title,
  on,
  toggle,
  className,
  disabled,
  icon,
  backgroundColor,
  fill,
  stroke,
  sideLeft,
  sideRight,
  options,
  right,
  active,
  onClick,
  dataQa,
}) => {
  const wrapper = cn('toolkit', styles.wrapper, className);

  const iconButton = cn(styles.iconButton, { [styles.rotate]: on });
  const [size, ref] = useMeasure();

  return (
    <div className={wrapper}>
      <Button
        ref={ref}
        onClick={toggle}
        className={styles.button}
        disabled={disabled}
        style={{ backgroundColor }}
        dataQa={dataQa}
      >
        {title}
        {icon && (
          <Icon
            component={ChevronRight}
            size="16"
            className={iconButton}
            fill={fill}
            stroke={stroke}
          />
        )}
      </Button>
      <OptionContainer
        on={on}
        toggle={toggle}
        size={size}
        options={options}
        right={right}
        className={className}
        onClick={onClick}
        active={active}
        sideLeft={sideLeft}
        sideRight={sideRight}
      />
    </div>
  );
};

Dropdown.displayName = 'Dropdown';

Dropdown.propTypes = {
  title: PropTypes.node,
  options: PropTypes.arrayOf(PropTypes.shape()).isRequired,
  on: PropTypes.bool.isRequired,
  onClick: PropTypes.func,
  right: PropTypes.bool,
  toggle: PropTypes.func,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  active: PropTypes.string,
  icon: PropTypes.bool,
  backgroundColor: PropTypes.string,
  sideLeft: PropTypes.bool,
  sideRight: PropTypes.bool,
  fill: PropTypes.string,
  stroke: PropTypes.string,
  dataQa: PropTypes.string,
};

Dropdown.defaultProps = {
  title: '',
  right: false,
  disabled: false,
  icon: true,
  backgroundColor: colors.balataGreen,
  className: undefined,
  sideLeft: false,
  sideRight: false,
  fill: colors.white,
  stroke: colors.white,
  dataQa: '',
  active: undefined,
  toggle: () => {},
  onClick: () => {},
};

export default Dropdown;
