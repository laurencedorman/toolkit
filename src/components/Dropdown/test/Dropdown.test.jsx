/**
 * Copyright (c) Colibri SAS - ManoMano
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { shallow } from 'enzyme';
import Button from '../../Button';
import Dropdown from '../Dropdown';
import Icon from '../../Icon';
import OptionContainer from '../OptionContainer';

describe('Dropdown', () => {
  const defaultProps = {
    options: [{ value: 'valueTest', label: 'labelTest', on: true }],
  };

  describe('template', () => {
    it('should match snapshot', () => {
      // When
      const wrapper = shallow(
        <Dropdown
          on
          title="titleTest"
          right
          disabled
          active="active"
          backgroundColor="#fff"
          className="my class name"
          sideLeft
          sideRight
          fill="#666"
          stroke="#555"
          dataQa="testQa"
          {...defaultProps}
        />
      );

      // Then
      expect(wrapper).toMatchSnapshot();
    });

    it('should match snapshot without default props', () => {
      // When
      const wrapper = shallow(<Dropdown {...defaultProps} />);

      // Then
      expect(wrapper).toMatchSnapshot();
    });

    it('should render no title and no icon when no props title and icon', () => {
      // When
      const wrapper = shallow(
        <Dropdown on={false} icon={false} {...defaultProps} />
      );

      // Then
      const button = wrapper.find(Button);
      expect(button.text()).toEqual('');
    });

    it('should render just an icon with no title', () => {
      // When
      const wrapper = shallow(<Dropdown on={false} {...defaultProps} />);

      // Then
      const icon = wrapper.find(Button).find(Icon);
      expect(icon).toHaveLength(1);
    });

    it('should render just title with no icon', () => {
      // When
      const wrapper = shallow(
        <Dropdown on={false} icon={false} title="titleTest" {...defaultProps} />
      );

      // Then
      const button = wrapper.find(Button);
      expect(button.text()).toEqual('titleTest');

      const icon = button.find(Icon);
      expect(icon).toHaveLength(0);
    });
  });

  describe('events', () => {
    describe('when button has been clicked', () => {
      it('should call toggle method', () => {
        // Given
        const onClickMock = jest.fn();
        const wrapper = shallow(
          <Dropdown on={false} toggle={onClickMock} {...defaultProps} />
        );
        const button = wrapper.find(Button);

        // When
        button.simulate('click');

        // Then
        expect(onClickMock).toHaveBeenCalledWith();
      });
    });

    describe('when option container has been clicked', () => {
      it('should call onClick method', () => {
        // Given
        const onClickMock = jest.fn();
        const wrapper = shallow(
          <Dropdown on={false} onClick={onClickMock} {...defaultProps} />
        );
        const optionContainer = wrapper.find(OptionContainer);

        // When
        optionContainer.simulate('click');

        // Then
        expect(onClickMock).toHaveBeenCalledWith();
      });
    });

    describe('when option container has been toggled', () => {
      it('should call toggle method', () => {
        // Given
        const toggleMock = jest.fn();
        const wrapper = shallow(
          <Dropdown on={false} toggle={toggleMock} {...defaultProps} />
        );

        const optionContainer = wrapper
          .find(OptionContainer)
          .dive()
          .find({ onClick: toggleMock });

        // When
        optionContainer.simulate('click');

        // Then
        expect(toggleMock).toHaveBeenCalledWith();
      });
    });
  });
});
