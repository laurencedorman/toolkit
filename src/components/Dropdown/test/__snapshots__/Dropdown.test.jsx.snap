// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown template should match snapshot 1`] = `
<div
  className="toolkit wrapper my class name"
>
  <Button
    className="button"
    dataQa="testQa"
    disabled={true}
    fill="#ffffff"
    onClick={[Function]}
    reverse={false}
    size="default"
    style={
      Object {
        "backgroundColor": "#fff",
      }
    }
    theme="default"
    title=""
    type="button"
  >
    titleTest
    <Icon
      className="iconButton rotate"
      component={[Function]}
      dataQa=""
      fill="#666"
      onClick={[Function]}
      onKeyPress={[Function]}
      size="16"
      stroke="#555"
    />
  </Button>
  <OptionContainer
    active="active"
    className="my class name"
    on={true}
    onClick={[Function]}
    options={
      Array [
        Object {
          "label": "labelTest",
          "on": true,
          "value": "valueTest",
        },
      ]
    }
    right={true}
    sideLeft={true}
    sideRight={true}
    size={
      Object {
        "bottom": 0,
        "height": 0,
        "left": 0,
        "right": 0,
        "top": 0,
        "width": 0,
        "x": 0,
        "y": 0,
      }
    }
    toggle={[Function]}
  />
</div>
`;

exports[`Dropdown template should match snapshot without default props 1`] = `
<div
  className="toolkit wrapper"
>
  <Button
    className="button"
    dataQa=""
    disabled={false}
    fill="#ffffff"
    onClick={[Function]}
    reverse={false}
    size="default"
    style={
      Object {
        "backgroundColor": "#29b9ad",
      }
    }
    theme="default"
    title=""
    type="button"
  >
    <Icon
      className="iconButton"
      component={[Function]}
      dataQa=""
      fill="#ffffff"
      onClick={[Function]}
      onKeyPress={[Function]}
      size="16"
      stroke="#ffffff"
    />
  </Button>
  <OptionContainer
    active=""
    className=""
    onClick={[Function]}
    options={
      Array [
        Object {
          "label": "labelTest",
          "on": true,
          "value": "valueTest",
        },
      ]
    }
    right={false}
    sideLeft={false}
    sideRight={false}
    size={
      Object {
        "bottom": 0,
        "height": 0,
        "left": 0,
        "right": 0,
        "top": 0,
        "width": 0,
        "x": 0,
        "y": 0,
      }
    }
    toggle={[Function]}
  />
</div>
`;
